src/main/java/com/waverleysoftware/pageobjects/GoogleAllPage.java

```diff
@@ -0,0 +1,31 @@
+package com.waverleysoftware.pageobjects;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static com.waverleysoftware.listemers.TestListener.getSoftAssert;
+
+/**
+ * A simple class which describes the behavior of google "all" page.
+ */
+public class GoogleAllPage extends BasePage {
+
+    private final List<WebElement> listOfLinks = findAll(By.cssSelector("#rso h3 > a"));
```

Comment:
in this case you attempt for find links, when page object is created... it's not a good practice... you should do that lazily...